Adding "empty" constraints after solve can trigger Gurobi.jl assertions

```julia
using JuMP, Gurobi
m = Model(solver=GurobiSolver())
@defVar(m, x, Bin)  # Just so there is something there
@addConstraint(m, 20 >= 0)
solve(m)  # fine
@addConstraint(m, 20 >= 0)
solve(m)  # error
```
```
julia> solve(m)
ERROR: AssertionError: (n_rows = num_constrs(m.inner)) == length(lb) == length(ub) == length(sense) == length(rhs)
 in updatemodel! at /Users/idunning/.julia/v0.4/Gurobi/src/GurobiSolverInterface.jl:170
 in setvartype! at /Users/idunning/.julia/v0.4/Gurobi/src/GurobiSolverInterface.jl:324
 in buildInternalModel at /Users/idunning/.julia/v0.4/JuMP/src/solvers.jl:343
 in solve at /Users/idunning/.julia/v0.4/JuMP/src/solvers.jl:81
```

With debug `@show`s added

```
julia> solve(m)  # error
length(num_constrs(m.inner)) = 1
length(lb) = 2
length(ub) = 2
length(sense) = 0
length(rhs) = 0
ERROR: AssertionError: (n_rows = num_constrs(m.inner)) == length(lb) == length(ub) == length(sense) == length(rhs)
 in updatemodel! at /Users/idunning/.julia/v0.4/Gurobi/src/GurobiSolverInterface.jl:175
 in setvartype! at /Users/idunning/.julia/v0.4/Gurobi/src/GurobiSolverInterface.jl:329
 in buildInternalModel at /Users/idunning/.julia/v0.4/JuMP/src/solvers.jl:343
 in solve at /Users/idunning/.julia/v0.4/JuMP/src/solvers.jl:81
```

`¯\_(ツ)_/¯`

--- a/test/probmod.jl
+++ b/test/probmod.jl
@@ -220,6 +220,23 @@ end
 end
 
 
+facts("[probmod] Test adding empty constraints") do
+for solver in lp_solvers
+context("With solver $(typeof(solver))") do
+    m = Model(solver=solver)
+    @defVar(m, 0 <= x <= 9)
+    @setObjective(m, Max, x)
+    @addConstraint(m, x <= 5)
+    @addConstraint(m, 0 <= 1)
+    solve(m)
+    @addConstraint(m, 0 <= 1)
+    @fact solve(m) --> :Optimal
+    @fact getValue(x) --> roughly(5.0, TOL)
+end
+end
+end
+
+
 facts("[probmod] Test bound modification on binaries") do
 for solver in ip_solvers
 context("With solver $(typeof(solver))") do

